tests/test_CFDataset.py

```diff
@@ -1,122 +1,253 @@
+    ('climo_gcm', 'unique_id', 'tasmax_msaClim_BNU-ESM_historical_r1i1p1_19650101-19701230'),
+], indirect=['tiny_dataset'])
+def test_simple_property(tiny_dataset, prop, expected):
+    assert getattr(tiny_dataset, prop) == expected
+
+
+@mark.parametrize('tiny_dataset, prop, expected', [
+    ('gcm', 'project', 'CMIP5'),
+    ('gcm', 'institution', 'BNU'),
+    ('gcm', 'model', 'BNU-ESM'),
+    ('gcm', 'emissions', 'historical'),
+    ('gcm', 'run', 'r1i1p1'),
+    
+    ('downscaled', 'project', 'CMIP5'),
+    ('downscaled', 'institution', 'PCIC'),
+    
+    ('hydromodel_gcm', 'project', 'CMIP5'),
+    ('hydromodel_gcm', 'institution', 'PCIC'),
+
+    ('climo_gcm', 'project', 'CMIP5'),
+    ('climo_gcm', 'institution', 'BNU'),
+    ('climo_gcm', 'model', 'BNU-ESM'),
+    ('climo_gcm', 'emissions', 'historical'),
+    ('climo_gcm', 'run', 'r1i1p1'),
+], indirect=['tiny_dataset'])
+def test_metadata_simple_property(tiny_dataset, prop, expected):
+    assert getattr(tiny_dataset.metadata, prop) == expected
+
+
+@mark.parametrize('tiny_dataset, expected', [
+    ('gcm', {'time', 'lon', 'lat', 'nb2'}),
+    ('downscaled', {'time', 'lon', 'lat'}),
+    ('hydromodel_gcm', {'time', 'lon', 'lat', 'depth'}),
+    ('climo_gcm', {'time', 'lon', 'lat', 'bnds'}),
+], indirect=['tiny_dataset'])
+def test_dim_names(tiny_dataset, expected):
+    assert set(tiny_dataset.dim_names()) == expected
+
+
+# Tests for some dimensions that are the same in all datasets.
+@mark.parametrize('tiny_dataset', [
+    'gcm',
+    'downscaled',
+    'hydromodel_gcm',
+    'climo_gcm',
+], indirect=True)
+@mark.parametrize('dim_name, expected', [
+    (['time'], {'T': 'time'}),
+    (['time', 'lon'], {'T': 'time', 'X': 'lon'}),
 ])
-def test_standard_climo_periods(calendar, key, start_date, end_date):
-    assert standard_climo_periods(calendar)[key] == (start_date, end_date)
-
-
-def test_first_MiB_md5sum(tiny_gcm):
-    assert tiny_gcm.first_MiB_md5sum == b'>\xb8\x12\xcc\xa96is\xb4\x10x\xb0\xbf\x19\xfe;'
-
-
-def test_dependent_varnames(tiny_gcm):
-    assert set(tiny_gcm.dependent_varnames) == {'tasmax'}
-
-
-def test_dim_names(tiny_gcm):
-    assert set(tiny_gcm.dim_names()) == {'time', 'lon', 'lat', 'nb2'}
-
-
-@mark.parametrize('dim_name, expected', (
-        (['time'], {'T': 'time'}),
-        (['time', 'lon'], {'T': 'time', 'X': 'lon'}),
-        (None, {'T': 'time', 'X': 'lon', 'Y': 'lat'})
-))
-def test_dim_axes_from_names(tiny_gcm, dim_name, expected):
-    assert tiny_gcm.dim_axes_from_names(dim_name) == expected
-
-
+def test_dim_axes_from_names(tiny_dataset, dim_name, expected):
+    assert tiny_dataset.dim_axes_from_names(dim_name) == expected
+
+
+# Tests for all dimensions - may differ between datasets.
+@mark.parametrize('tiny_dataset, expected', [
+    ('gcm', {'T': 'time', 'X': 'lon', 'Y': 'lat'}),
+    ('downscaled', {'T': 'time', 'X': 'lon', 'Y': 'lat'}),
+    ('hydromodel_gcm', {'T': 'time', 'X': 'lon', 'Y': 'lat'}),  # why isn't depth in this list?
```

Comment:
Though VIC models multiple depths, I am pretty sure that we don't save multi-depth output.